src/restrictingGuards.ts

```diff
@@ -2,6 +2,16 @@ import {checkerToGuard} from './Checker';
 import {andGuard, orGuard, thenGuard} from './Combinators';
 import {ReasonGuard} from './ReasonGuard';
 import {isNumber, isSymbol, isString} from './primitiveGuards';
+import {isDate} from './instanceGuards';
+import {isDateString} from './parseGuards';
+
+export const isDateOrStringDate = orGuard(
```

Comment:
Given the `parseGuards` bit is called `isDateString`, I think this should be called `isDateOrDateString` ... that also parses better as English IMO :)